### DISCUSSION TITLE
Who is using Pebble?

### DISCUSSION THREAD

**Discussion by smiklosovic** [PebbleTemplates/pebble#737] [General]
Hi,

I like this project and I would like to know what sites are using Pebble. Do you have some list of sites which are backed by this template engine?

Thanks!

**Comment by mbosecke** [comment]
Unfortunately I don't have a list! I wish I did!

**Comment by jcarvalho** [comment]
We at FenixEdu are using Pebble for our theming engine for our FenixEdu application family. [Source here - bennu-portal subproject](https://github.com/FenixEdu/bennu).

We are also using it as the main engine for our Java-based CMS: https://github.com/FenixEdu/fenixedu-cms.

**Comment by decebals** [comment]
Pebble is used in [Pippo](https://github.com/decebals/pippo). I and [James](https://github.com/gitblit) use Pebble as main template engine in our projects. Nice library, thanks!

**Comment by realcnbs** [comment]
We are using pebble in a ~100 of celebrity blog sites with relatively high traffic.

**Comment by pacey** [comment]
Hey I work at Rightmove.co.uk and we have just switched over to using Pebble instead of Handlebars.java for the new search results page. Good work! Feel free to contact me if you'd like to know more

**Comment by capnfabs** [comment]
We're using it in an Android app in http://github.com/projectbuendia/client

**Comment by mbosecke** [comment]
I love hearing about the different projects using Pebble. I'm going to keep this issue open indefinitely for people to continue sharing. Thanks guys.

**Comment by ebussieres** [comment]
I'm currently using Pebble in a spring mvc application for a transactional website. I was using Thymeleaf and just switched to Pebble mainly due by performance issues with Thymeleaf. Rendering is so much faster with Pebble (about 7 times faster), I love it.

**Comment by jcarvalho** [comment]
We are also using Pebble to implement our own version of Spring's [Initializr](https://github.com/spring-io/initializr), also known as [start.spring.io](https://start.spring.io).

The tool generates a Maven project based on some user input, allowing the user to customize the package names, artifact ids, chosen modules, etc. I found this format easier to use than regular Maven archetypes, with a more user-friendly Web UI.

With Pebble, we can easily specify the project structure using Pebble expressions, both for file names and file contents. An example of such a file can be found [here](https://github.com/FenixEdu/fenixedu-start/blob/master/src/main/resources/templates/module/src/main/java/%7B%7Brequest.packageName%20%7C%20path%7D%7D/servlet/%7B%7Brequest.upperCamelCaseName%7D%7DInitializer.java).

Full Source code available [here](https://github.com/FenixEdu/fenixedu-start).

**Comment by derekvandaal** [comment]
We are using Pebble as the template engine to render emails being sent out
notifying customers of any movement on their accounts at one of the big 4
banks in South Africa.

This volumes are huge and Pebble handles the load with ease :)

**Comment by hectorlf** [comment]
As of today, my blogging app uses Pebble for its front. LOL.

And I'm happy to bring this measurements: the combination of Stripes Framework-Pebble performs 2-3 times faster than Stripes Framework-JSP. That was totally unexpected. Well done, Mitchell!

And next in line, I'd like to integrate Pebble with Spring Boot. Is anyone at it already?

**Comment by rzhb** [comment]
@hectorlf 
[Spring Boot Starter Pebble](https://github.com/LionelWoody/spring-boot-starter-pebble) is the only one that I found integrate Pebble with Spring Boot.

**Comment by hectorlf** [comment]
@zhibinr oh, did a search on github and that project didn't come up. It's a pity, because i've already got mine working :-/

**Comment by ablipan** [comment]
We are using Pebble as the backend template. [101test.com](http://101test.com)

**Comment by dupirefr** [comment]
I just discover Pebble and decided to use it with JavaFX. As a matter of fact FXML files have really limited features and Pebble is perfect to overcome the weaknesses of FXML.

Thanks a lot for this library!

**Comment by jochen777** [comment]
We are using it for an upcoming service at a big german website. Very handy and useful. And speedy :) We love the "compatibility" with twig, because we have a PHP team as well. So the fronted-teams can share knowledge.

**Comment by hectorlf** [comment]
Starting next week, it's going to be used in an internal project of a Dublin based company. Hopefully, more in the future.

**Comment by MrPowerGamerBR** [comment]
I'm currently using Sparkjava + Pebble for my Minecraft Server website rewrite.

After using Sparkjava + Pebble, I doubt I will go back to writing PHP websites again... it is just so good.

**Comment by MrPowerGamerBR** [comment]
I forgot to update, now my website using Pebble is live! 🙌 http://sparklypower.net/

(Yeah, it is live but I still didn't come up with something cool to add to the homepage, sorry!)

And soon my personal website http://mrpowergamerbr.com will also use Pebble.

If anyone is curious: I used Sparkjava + MongoDB + Pebble.

Pebble is something that just "works", and, after you learn to use it, you won't be able to go back.

**Comment by donnior** [comment]
We started using pebble 1.x from last year in our project, now it serves ~5000 users every day and works very well.

**Comment by kshep92** [comment]
Pebble is the templating language of choice for all web based information systems at my government office. Thanks so much for this library!

**Comment by MrPowerGamerBR** [comment]
Almost 6 months ago I said I started using Pebble.

And I'm still using Pebble lol, it is a such awesome template engine, sometimes I'm like "there is no way this code will work" and bam, Pebble throws my pessimism out of the window and proves that yes, this code that I thought that Pebble would crap out really works.

Pebble Template Engine: It just works:tm:

(Here's another website I made that uses Pebble Template Engine: https://loritta.website/)

**Comment by zooooooooy** [comment]
I try find a template engine to use for my system. Meet my project. After trying, I am satisfied with the efficiency and grammar habits.

**Comment by lccanh** [comment]
I am going to try this out, thanks

**Comment by cjbrooks12** [comment]
I have been using Pebble as the default template engine for my new project, Orchid, which I'm taking public today! Pebble is such a pleasure to work with, and it's so _fast_, I couldn't have made Orchid without Pebble!

https://github.com/JavaEden/Orchid

**Comment by electrotype** [comment]
Pebble is the main templating engine in Spincast Framework:
https://www.spincast.org/documentation#templating_engine

A big ❤️ from us!

**Comment by flef** [comment]
Hi, I'm using Pebble to generate Word docx document or PDF. I designed them directly into word to ensure I have a nice result.

See https://github.com/flef/Webble.

**Comment by stevespringett** [comment]
OWASP Dependency-Track uses Pebble for transforming system and portfolio alerts into Slack and Microsoft Teams inbound webhooks, general-purpose webhooks, email notifications, and output to console.

**Comment by colesico** [comment]
We use pebble as a main template engine in  colesico-framework
[Colesico application microframework](https://github.com/colesico/colesico-framework)

**Comment by umesh2k8** [comment]
I am working in company to generate the emails, We use Pebble template to generate the dynamic content for my customer emails. It is very good template engine and help to extend the functionalities for our requirement. I love Pebble.
